geode-core/src/main/java/org/apache/geode/internal/offheap/MemoryAllocatorImpl.java

```diff
@@ -360,6 +394,9 @@ public void close() {
     try {
       LifecycleListener.invokeBeforeClose(this);
     } finally {
+      if (updateStatsTask != null) {
```

Comment:
realClose should be changed to have a call of shutdown on your new "final" ScheduledExecutorService. You could keep a reference to the ScheduledFuture returned by scheduleWithFixedDelay and call cancel() on it here. The complication is that we don't always return the offheap memory to the OS to prevent SEGV and this can cause us to reuse that offheap memory when doing a reconnect. See the MemoryAllocatorImpl.reuse method